srtcore/sync.h

```diff
@@ -536,23 +537,47 @@ class CSync
     }
 
     // Static ad-hoc version
-    static void lock_signal(Condition& cond, Mutex& m)
+    SRT_ATR_DEPRECATED_PX static void lock_signal(Condition& cond, Mutex& m) SRT_ATR_DEPRECATED
     {
         ScopedLock lk(m); // XXX with thread logging, don't use ScopedLock directly!
         cond.notify_one();
     }
 
-    static void lock_broadcast(Condition& cond, Mutex& m)
+    static void lock_notify_one(Condition& cond, Mutex& m)
+    {
+        ScopedLock lk(m); // XXX with thread logging, don't use ScopedLock directly!
+        cond.notify_one();
+    }
+
+
+    SRT_ATR_DEPRECATED_PX static void lock_broadcast(Condition& cond, Mutex& m) SRT_ATR_DEPRECATED
+    {
+        ScopedLock lk(m); // XXX with thread logging, don't use ScopedLock directly!
+        cond.notify_all();
+    }
+    static void lock_notify_all(Condition& cond, Mutex& m)
     {
         ScopedLock lk(m); // XXX with thread logging, don't use ScopedLock directly!
         cond.notify_all();
     }
 
-    void signal_locked(UniqueLock& lk SRT_ATR_UNUSED)
+
+    SRT_ATR_DEPRECATED_PX  void signal_locked(UniqueLock& lk SRT_ATR_UNUSED) SRT_ATR_DEPRECATED
```

Comment:
What is the benefit of deprecating an internally used function over just removing/replacing it with the new one?

Reply:
Actually I did it as the first thing in order to detect each use of this function. I was unsure if there isn't any other place where it is used. Likely can be removed.